src/browser/stacktrace/utils.ts

```diff
@@ -158,21 +158,48 @@ export const filterExtraWdioFrames = (error: Error): Error => {
             return error;
         }
 
-        const isWdioFrame = (frame: StackFrame): boolean => {
-            return Boolean(frame.fileName && frame.fileName.includes("/node_modules/webdriverio/"));
-        };
+        // If we found something more relevant (e.g. user's code), we can remove useless testplane internal frames
+        // If we haven't, we keep testplane internal frames
+        const shouldDropTestplaneInternalFrames =
+            getStackTraceRelevance(error) > FRAME_RELEVANCE.projectInternals.value;
 
-        const isIgnoredFunction = (frame: StackFrame): boolean => {
-            const funcName = frame.functionName;
+        const isIgnoredWebdriverioFrame = (frame: StackFrame): boolean => {
+            const isWebdriverioFrame = frame.fileName && frame.fileName.includes("/node_modules/webdriverio/");
+            const fnName = frame.functionName;
 
-            if (!funcName) {
+            if (!isWebdriverioFrame || !fnName) {
                 return false;
             }
 
-            return Boolean(WDIO_IGNORED_STACK_FUNCTIONS.some(fn => fn === funcName || "async " + fn === funcName));
+            return WDIO_IGNORED_STACK_FUNCTIONS.some(fn => fn === fnName || "async " + fn === fnName);
+        };
+
+        const isWdioUtilsFrame = (frame: StackFrame): boolean => {
+            return Boolean(frame.fileName && frame.fileName.includes("/node_modules/@wdio/utils/"));
         };
 
-        const shouldIncludeFrame = (frame: StackFrame): boolean => !isWdioFrame(frame) || !isIgnoredFunction(frame);
+        const isTestplaneExtraInternalFrame = (frame: StackFrame): boolean => {
+            const testplaneExtraInternalFramePaths = [
+                "/node_modules/testplane/src/browser/history/",
+                "/node_modules/testplane/src/browser/stacktrace/",
+            ];
+
+            return Boolean(
+                frame.fileName && testplaneExtraInternalFramePaths.some(path => frame.fileName?.includes(path)),
+            );
+        };
+
+        const shouldIncludeFrame = (frame: StackFrame): boolean => {
+            if (isIgnoredWebdriverioFrame(frame) || isWdioUtilsFrame(frame)) {
```

Comment:
why should we separate these checks?

Reply:
One is webdriverio frame, and other is wdioUtils frame.
For webdriverio frames we only ignore some but not all frames
I think its just easier to read when we have multiple structured conditions instead of one function with lots of "||"